File: Resources programmatically/MainActivity.java
package com.example.myapplication;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;
import androidx.core.content.res.ResourcesCompat;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(new View(this));
        /**/
        init();
    }

    private void init() {
        //  Get color for TextView
        int textColor = ContextCompat.getColor(this, android.R.color.white);
        //  Get color background for Views
        Drawable backgroundColorNoTheme = ResourcesCompat.getDrawable(getResources(), android.R.color.white, null);
        //  Get color background for Views
        Drawable backgroundColorWithTheme = ResourcesCompat.getDrawable(getResources(), android.R.color.white, getTheme());
        //  A) drawables with theme attributes
        Drawable drawable = ContextCompat.getDrawable(this, android.R.drawable.btn_star);
        //  B) drawables without theme attributes
        Drawable drawableNoTheme = ResourcesCompat.getDrawable(getResources(), android.R.drawable.btn_star, null);
        //  C) drawables with theme attributes from another theme
        Drawable drawableWithTheme = ResourcesCompat.getDrawable(getResources(), android.R.drawable.btn_star, getTheme());
    }
}

File: Resources programmatically/MainActivity.kt
package com.example.myapplication

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.core.content.res.ResourcesCompat

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(View(this))
        /**/
        init()
    }

    private fun init() {
        //  Get color for TextView
        val textColor = ContextCompat.getColor(this, android.R.color.white)
        //  Get color background for Views
        val backgroundColorNoTheme = ResourcesCompat.getDrawable(resources, android.R.color.white, null)
        //  Get color background for Views
        val backgroundColorWithTheme = ResourcesCompat.getDrawable(resources, android.R.color.white, theme)
        //  A) drawables with theme attributes
        val drawable = ContextCompat.getDrawable(this, android.R.drawable.btn_star)
        //  B) drawables without theme attributes
        val drawableNoTheme = ResourcesCompat.getDrawable(resources, android.R.drawable.btn_star, null)
        //  C) drawables with theme attributes from another theme
        val drawableWithTheme = ResourcesCompat.getDrawable(resources, android.R.drawable.btn_star, theme)
    }
}
